File: backend/src/main/java/com/igreja/site/controller/ServicoController.java
package com.igreja.site.controller;

import com.igreja.site.model.Servico;
import com.igreja.site.repository.ServicoRepository;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/servicos")
public class ServicoController {

    private final ServicoRepository repository;

    public ServicoController(ServicoRepository repository) {
        this.repository = repository;
    }

    @GetMapping
    public List<Servico> listar() {
        return repository.findAll();
    }

    @PostMapping
    public ResponseEntity<Servico> criar(@Valid @RequestBody Servico servico) {
        return ResponseEntity.status(201).body(repository.save(servico));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> excluir(@PathVariable Long id) {
        if (!repository.existsById(id)) {
            return ResponseEntity.notFound().build();
        }
        repository.deleteById(id);
        return ResponseEntity.noContent().build();
    }
}

File: backend/src/controllers/servicoController.js
const { Servico } = require('../models');

async function listar(req, res) {
  const servicos = await Servico.findAll();
  return res.json(servicos);
}

async function criar(req, res) {
  const servico = await Servico.create(req.body);
  return res.status(201).json(servico);
}

async function excluir(req, res) {
  const servico = await Servico.findByPk(req.params.id);
  if (!servico) return res.status(404).end();
  await servico.destroy();
  return res.status(204).end();
}

module.exports = { listar, criar, excluir };
